reroute/src/Web/Routing/Combinators.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+module Web.Routing.Combinators where
+
+import Data.HVect (Append)
+import Data.String
+import Data.Typeable (Typeable)
+import Web.PathPieces
+import qualified Data.Text as T
+
+import Web.Routing.SafeRouting
+
+data PathState = Open | Closed
+
+data SafePath (as :: [*]) (pathState :: PathState) where
+  SP_Empty :: SafePath '[] 'Open
+  SP_StaticCons :: T.Text -> SafePath as ps -> SafePath as ps
+  SP_VarCons :: (PathPiece a, Typeable a) => SafePath as ps -> SafePath (a ': as) ps
+  SP_Wildcard :: SafePath as 'Open -> SafePath (T.Text ': as) 'Closed
+
+toUnsafePath :: SafePath as pathState -> Path as
+toUnsafePath SP_Empty = Empty
+toUnsafePath (SP_StaticCons t p) = StaticCons t (toUnsafePath p)
+toUnsafePath (SP_VarCons p) = VarCons (toUnsafePath p)
+toUnsafePath (SP_Wildcard p) = Wildcard (toUnsafePath p)
+
+type Var a = SafePath (a ': '[]) 'Open
```

Comment:
You are correct: it is not used throughout the library. However, it might be used in apps built with Spock, since its usage suggested in https://www.spock.li/2015/04/19/type-safe_routing.html